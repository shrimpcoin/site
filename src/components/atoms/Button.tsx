import styled from 'styled-components';

type ButtonProps = {
  background: string;
  outline: string;
};

export const Button = styled.a<ButtonProps>`
  display: inline-block;
  appearance: button;
  -moz-appearance: button;
  -webkit-appearance: button;
  text-align: center;
  text-decoration: none;
  font-family: 'Manrope';
  font-weight: 200;
  font-size: 15px;
  vertical-align: middle;
  padding: 1em 2em;
  cursor: pointer;

  transition: all 0.2s ease;
  box-sizing: border-box;

  background: ${(props) => props.background};
  border: 1px solid ${(props) => props.outline};
  color: ${(props) => props.outline};

  &:hover {
    background: ${(props) => props.outline};
    border: 1px solid ${(props) => props.background};
    color: ${(props) => props.background};
    transition: 0.3s ease-out;
  }
`;
