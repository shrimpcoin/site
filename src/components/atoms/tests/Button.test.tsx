import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

test('Button', () => {
  const component = renderer.create(<Button href="#">Button</Button>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
