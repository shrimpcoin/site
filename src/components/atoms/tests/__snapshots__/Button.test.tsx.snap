// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<a
  className="sc-bdnxRM hhsNFu"
  href="#"
>
  Button
</a>
`;
